hera_cal/redcal.py

```diff
@@ -147,6 +150,71 @@ def get_reds(antpos, pols=['xx'], pol_mode='1pol', ex_ants=[], bl_error_tol=1.0,
     return add_pol_reds(pos_reds, pols=pols, pol_mode=pol_mode, ex_ants=ex_ants)
 
 
+def filter_reds(reds, bls=None, ex_bls=None, ants=None, ex_ants=None, ubls=None, ex_ubls=None, pols=None, ex_pols=None):
+    '''
+    Filter redundancies to include/exclude the specified bls, antennas, unique bl groups and polarizations.
+    Assumes reds indices are Antpol objects.
+    Args:
+        reds: list of lists of redundant baseline tuples, e.g. (ind1,ind2,pol)
+        bls (optional): baselines as antenna pair tuples (i,j) to include in reds.
+        ex_bls (optional): baselines as antenna pair tuples (i,j) to exclude in reds.
+        ants (optional): antenna numbers (as int's) to include in reds.
+        ex_ants (optional): antenna numbers (as int's) to exclude in reds.
+        ubls (optional): baselines representing their redundant group to include in reds.
+        ex_ubls (optional): baselines representing their redundant group to exclude in reds.
+        pols (optional): polarizations to include in reds. e.g. 'xy' or 'yx'.
+        ex_pols (optional): polarizations to exclude in reds. e.g. 'xy' or 'yx'.
+    Return:
+        reds: list of lists of redundant baselines as antenna pair tuples.
+    '''
+    if pols is None: # if no pols are provided, deduce them from the reds
+        pols = set(gp[0][2] for gp in reds)
```

Comment:
This isn't quite as general as it could be: in the case where two different pols are considered redundant, this doesn't necessarily get all pols.

I'd instead do `pols = set([bl[2] for gp in reds for bl in gp])`.